app/bundles/CoreBundle/Twig/Extension/ThemesExtension.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Mautic\CoreBundle\Twig\Extension;
+
+use Mautic\CoreBundle\Helper\CoreParametersHelper;
+use Twig\Extension\AbstractExtension;
+use Twig\TwigFunction;
+
+final class ThemesExtension extends AbstractExtension
+{
+    public function __construct(private CoreParametersHelper $coreParametersHelper)
+    {
+    }
+
+    public function getFunctions()
+    {
+        return [
+            new TwigFunction('theme_text_on_color', [$this, 'theme_text_on_color']),
+            new TwigFunction('theme_text_on_color__helper', [$this, 'theme_text_on_color__helper']),
+            new TwigFunction('theme_primary_color', [$this, 'theme_primary_color']),
```

Comment:
I don't want to propose changes that you'll have to make in all the Twig files, but can we replace the 3 `_` with just one in `theme_text_on_color__helper`? Or consider some more descriptive name? I don't know what to expect from such twig function.
```suggestion
            new TwigFunction('theme_text_on_color', [$this, 'themeTextOnColor']),
            new TwigFunction('theme_text_on_color__helper', [$this, 'themeTextOnColorHelper']),
            new TwigFunction('theme_primary_color', [$this, 'themePrimaryColor']),
```